routes/projects/update.js

```diff
@@ -0,0 +1,28 @@
+/*
+ * PUT /projects/:username/:projectName
+ * Creates a new project for the given user. A project object is sent in
+ * the form:
+ * {
+ *   name: <string> required,
+ *   username: <string> required,
+ *   isPublic: <boolean> required,
+ *   publishedAt: <dataTime> optional,
+ *   publishLink: <string> optional,
+ *   files: [ <FileObj1>, <FileObj2>... ] required
```

Comment:
Since a foreign key won't exist yet, my vote is to keep it simple - Creating a project means it has no files associated with it yet. Adding files through the `file` routes can update the related `project` object.